jax/lax.py

```diff
@@ -360,11 +360,77 @@ def clamp(min, x, max):
   return clamp_p.bind(min, x, max)
 
 def concatenate(operands, dimension):
+  """Concatenates a sequence of arrays along `dimension`.
+
+  Wraps XLA's `Concatenate
+  <https://www.tensorflow.org/xla/operation_semantics#concatenate>`_
+  operator.
+
+  Args:
+    operands: a sequence of arrays to concatenate. The arrays must have equal
+      shapes, except in the `dimension` axis.
+    dimension: the dimension along which to concatenate the arrays.
+
+  Returns:
+    An array containing the concatenation.
+  """
   return concatenate_p.bind(*operands, dimension=dimension,
                             operand_shapes=tuple(o.shape for o in operands))
 
 def conv_general_dilated(lhs, rhs, window_strides, padding, lhs_dilation=None,
                          rhs_dilation=None, dimension_numbers=None):
+  """General n-dimensional convolution operator, with optional dilation.
+
+  Wraps XLA's `Conv
+  <https://www.tensorflow.org/xla/operation_semantics#conv_convolution>`_
+  operator.
+
+  Args:
+    lhs: a rank `n+2` dimensional input array.
+    rhs: a rank `n+2` dimensional array of kernel weights.
+    window_strides: a sequence of `n` integers, representing the inter-window
+      strides.
+    padding: either the string `'SAME'`, the string `'VALID'`, or a sequence of
+      `n` `(low, high)` integer pairs that give the padding to apply before and
+      after each spatial dimension.
+    lhs_dilation: `None`, or a sequence of `n` integers, giving the
+      dilation factor to apply in each spatial dimension of `lhs`. LHS dilation
+      is also known as transposed convolution.
+    rhs_dilation: `None`, or a sequence of `n` integers, giving the
+      dilation factor to apply in each spatial dimension of `rhs`. RHS dilation
+      is also known as atrous convolution.
+    dimension_numbers: either `None`, a `ConvDimensionNumbers` object, or
+      a 3-tuple `(lhs_spec, rhs_spec, out_spec)`, where each element is a string
+      of length `n+2`.
+
+  Returns:
+    An array containing the convolution result.
+
+  In the string case of `dimension_numbers`, each character identifies by
+  position:
+
+  - the batch dimensions in `lhs`, `rhs`, and the output with the character
+    'N',
+  - the feature dimensions in `lhs` and the output with the character 'C',
+  - the input and output feature dimensions in rhs with the characters 'I'
+    and 'O' respectively, and
+  - spatial dimension correspondences between lhs, rhs, and the output using
+    any distinct characters.
+
+  For example, to indicate dimension numbers
```

Comment:
Text wrapping? Or maybe this is intentional.

Reply:
Done.